pkg/clusteragent/autoscaling/workload/provider.go

```diff
@@ -0,0 +1,50 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build kubeapiserver
+
+package workload
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/DataDog/datadog-agent/pkg/clusteragent/autoscaling"
+	"github.com/DataDog/datadog-agent/pkg/clusteragent/autoscaling/workload/model"
+	"github.com/DataDog/datadog-agent/pkg/util/kubernetes/apiserver"
+	"github.com/DataDog/datadog-agent/pkg/util/kubernetes/apiserver/leaderelection"
+)
+
+// StartWorkloadAutoscaling starts the workload autoscaling controller
+func StartWorkloadAutoscaling(ctx context.Context, apiCl *apiserver.APIClient, rcClient rcClient) error {
```

Comment:
Same as before. Here we don't have a retry mechanism and the error is simply logged. Are we fine with this behavior ?

Reply:
Can we do it in parallel ? I suppose we could have a goroutine to retry starting the autoscaler periodically

Reply:
I think I'd prefer CLB for visibility